openmdao/recorders/tests/test_list_vars.py

```diff
@@ -0,0 +1,299 @@
+
+        case = om.CaseReader('list_vars.db').get_case(0)
+
+        io_vars = case.list_vars(units=True, out_stream=None, return_format='dict')
+
+        self.assertEqual(io_vars, expected)
+
+    def test_list_vars_tags(self):
+        p = om.Problem()
+        model = p.model
+
+        model.add_recorder(om.SqliteRecorder('addsubtags.db'))
+
+        nn = 1
+        ivc = om.IndepVarComp()
+        ivc.add_output(name='a', shape=(nn,))
+        ivc.add_output(name='b', shape=(nn,))
+
+        model.add_subsystem(name='ivc', subsys=ivc,
+                            promotes_outputs=['a', 'b'])
+
+        adder = model.add_subsystem(name='add_subtract_comp', subsys=om.AddSubtractComp())
+        adder.add_equation('adder_output', ['input_a','input_b'], tags={'foo'})
+        adder.add_equation('adder_output2', ['input_a','input_a'], tags={'bar'})
+
+        model.connect('a', 'add_subtract_comp.input_a')
+        model.connect('b', 'add_subtract_comp.input_b')
+
+        p.setup()
+
+        p['a'] = np.random.rand(nn,)
+        p['b'] = np.random.rand(nn,)
+
+        p.run_model()
+
+        case = om.CaseReader('addsubtags.db').get_case(0)
+
+        a = p['a']
+        b = p['b']
+
+        for obj in (model, case):
+            foo_outputs = obj.list_vars(tags={'foo'}, print_tags=True,
+                                        out_stream=None, return_format='dict')
+            self.assertEqual(len(foo_outputs), 1,
+                             msg=f"There should be one output tagged 'foo': {foo_outputs}")
+
+            bar_outputs = obj.list_vars(tags={'bar'}, print_tags=False,
+                                        out_stream=None, return_format='dict')
+            self.assertEqual(len(bar_outputs), 1,
+                             msg=f"There should be one output tagged 'bar': {bar_outputs}")
+
+            assert_near_equal(foo_outputs['add_subtract_comp.adder_output']['val'], a + b)
+            assert_near_equal(bar_outputs['add_subtract_comp.adder_output2']['val'], a + a)
+
+    def test_list_vars_bool_args(self):
+        from openmdao.test_suite.components.sellar_feature import SellarMDAWithUnits
+
+        model = SellarMDAWithUnits()
+        model.add_subsystem('indep', om.IndepVarComp('foo', 1.0))
+        model.add_design_var('z', lower=np.array([-10.0, 0.0]),
+                                  upper=np.array([10.0, 10.0]))
+        model.add_design_var('x', lower=0.0, upper=10.0)
+        model.add_objective('obj')
+        model.add_constraint('con1', upper=0.0)
+        model.add_constraint('con2', upper=0.0)
+
+        prob = om.Problem(model)
+        prob.setup()
+
+        model.add_recorder(om.SqliteRecorder('list_vars.db'))
+
+        prob.set_solver_print(0)
+        prob.run_model()
+        prob.cleanup()
+
+        case = om.CaseReader('list_vars.db').get_case(0)
+
+        for arg in (
+            'val',
+            'prom_name',
+            'residuals',
+            # 'residuals_tol=None,
```

Comment:
I left them there on purpose, as documentation that those args were not being tested here because they are not boolean.

I can remove them or add a comment to clarify, whichever...

Reply:
I updated the test to use a  more robust method of identifying boolean args and testing with the non-default value